Kafka/KafkaJCAAPI/src/main/java/fish/payara/cloud/connectors/kafka/inbound/KafkaAsynchWorker.java

```diff
@@ -127,24 +138,28 @@ public void release() {
 
     @Override
     public void run() {
-        ConsumerRecords<Object, Object> records = consumer.poll(Duration.of(key.getSpec().getPollInterval(), ChronoUnit.MILLIS));
-        if (!records.isEmpty()) {
-            try {
-                // ok we got something
-                // schedule on the Work Manager Again
-                wm.scheduleWork(new RecordsProcessor(records));
-            } catch (WorkException ex) {
-                Logger.getLogger(KafkaAsynchWorker.class.getName()).log(Level.SEVERE, null, ex);
-            }
-            
-            if (key.getSpec().getCommitEachPoll()) {
-                consumer.commitSync();
+        if (!started.get()) {
+            checkStart();
```

Comment:
this is a busy wait. Can you sleep for the remaining delay?

Reply:
should be better now